caluma/caluma_workflow/domain_logic.py

```diff
@@ -528,3 +528,66 @@ def post_resume(work_item, user, context=None):
         )
 
         return work_item
+
+
+class RedoWorkItemLogic:
+    @classmethod
+    def validate_for_redo(cls, work_item):
+        if work_item.status == models.WorkItem.STATUS_READY:
```

Comment:
I think this needs to be documented. I was struggling with verifying the correctness here. Something along the lines of

> Validate if we can redo (jump back to) the given work item.
> Every task has a potential JEXL list of other tasks that can be redone from that task's workitem (redoable).
> We allow redoing a work item if at least one work item that came after this one allows "our" task to be re-done.

Reply:
Done, but it's not exactly as u wrote.

```
        Validate if we can redo (jump back to) the given work item.

        Every taskflow has a potential JEXL list of other tasks that can be redone from
        that task's next workitem (redoable).
        We allow redoing a work item if at least one ready work item allows "our" task
        to be re-done.
```